src/main/kotlin/org/rust/lang/core/psi/RsPsiFileInternals.kt

```diff
@@ -0,0 +1,32 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.lang.core.psi
+
+import com.google.common.annotations.VisibleForTesting
+import com.intellij.psi.impl.source.PsiFileImpl
+import com.intellij.psi.stubs.PsiFileStub
+import org.rust.openapiext.isUnitTestMode
+import java.lang.reflect.Method
+
+object RsPsiFileInternals {
+    @VisibleForTesting
+    val setStubTreeMethod: Method? = try {
+        PsiFileImpl::class.java
+            .getDeclaredMethod("setStubTree", PsiFileStub::class.java)
+            .apply { isAccessible = true }
+    } catch (e: Throwable) {
+        if (isUnitTestMode) throw e else null
+    }
```

Comment:
If the method will become unavailable in the future, the "slow" implementation without stubs will be used automatically, and also we'll detect this regression by failing tests